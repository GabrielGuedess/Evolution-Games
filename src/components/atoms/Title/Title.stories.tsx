import { Story, Meta } from '@storybook/react/types-6-0';

import { Title, TitleProps } from './Title';

export default {
  title: 'Atoms/Title',
  component: Title,
  args: {
    children: 'Bestsellers',
    textAlign: 'center',
  },
} as Meta<TitleProps>;

export const Desktop: Story<TitleProps> = args => (
  <div style={{ textAlign: 'center' }}>
    <Title {...args} />
  </div>
);

export const Mobile: Story<TitleProps> = args => (
  <div style={{ textAlign: 'center' }}>
    <Title {...args} />
  </div>
);

Mobile.parameters = {
  viewport: {
    defaultViewport: 'mobile1',
  },
};
