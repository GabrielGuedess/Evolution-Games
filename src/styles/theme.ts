export default {
  grid: {
    container: '165.6rem',
  },
  border: {
    radius: {
      regular: '1rem',
    },
  },
  font: {
    family: {
      primary:
        "'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    },
    regular: 400,
    medium: 500,
    sizes: {
      xxsmall: '1rem',
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      xmedium: '1.8rem',
      xxmedium: '2.0rem',
      big: '2.2rem',
      xbig: '2.5rem',
      xxbig: '2.8rem',
      xxxbig: '3.0rem',
      large: '3.2rem',
      xlarge: '4.5rem',
      xxlarge: '5.0rem',
      extra: '6.0rem',
    },
  },
  colors: {
    primary: '#7C3AED',
    secondary: '#717B8A',
    mainBg: '#04040C',
    white: '#FFFFFF',
    whiteText: '#F4F4F5',
    gameDetails: '#010914',
    gameGreat: '#0FBF00',
    gameAverage: '#FFF720',
    gameBad: '#FF3333',
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem',
    big: '5.8rem',
    huge: '8.0rem',
    section: '10rem',
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },
  shadows: {
    text: '0 0.8rem 1rem rgba(118, 74, 241, 0.5)',
    box: '0 0.8rem 2.8rem rgba(118, 74, 241, 0.5)',
  },
} as const;
