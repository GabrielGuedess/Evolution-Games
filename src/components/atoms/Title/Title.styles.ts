import media from 'styled-media-query';

import styled, { css, DefaultTheme, keyframes } from 'styled-components';

import { TitleProps } from './Title';

const lights = keyframes`
  0% {
    color: hsl(230, 40%, 80%);
    text-shadow:
      0 0 1em hsla(320, 100%, 50%, 0.2),
      0 0 0.125em hsla(320, 100%, 60%, 0.3),
      -1em -0.125em 0.5em hsla(40, 100%, 60%, 0),
      1em 0.125em 0.5em hsla(200, 100%, 60%, 0);
  }

  30% {
    color: hsl(230, 80%, 90%);
    text-shadow:
      0 0 1em hsla(320, 100%, 50%, 0.5),
      0 0 0.125em hsla(320, 100%, 60%, 0.5),
      -0.5em -0.125em 0.25em hsla(40, 100%, 60%, 0.2),
      0.5em 0.125em 0.25em hsla(200, 100%, 60%, 0.4);
  }

  40% {
    color: hsl(230, 100%, 95%);
    text-shadow:
      0 0 1em hsla(320, 100%, 50%, 0.5),
      0 0 0.125em hsla(320, 100%, 90%, 0.5),
      -0.25em -0.125em 0.125em hsla(40, 100%, 60%, 0.2),
      0.25em 0.125em 0.125em hsla(200, 100%, 60%, 0.4);
  }

  70% {
    color: hsl(230, 80%, 90%);
    text-shadow:
      0 0 1em hsla(320, 100%, 50%, 0.5),
      0 0 0.125em hsla(320, 100%, 60%, 0.5),
      0.5em -0.125em 0.25em hsla(40, 100%, 60%, 0.2),
      -0.5em 0.125em 0.25em hsla(200, 100%, 60%, 0.4);
  }

  100% {
    color: hsl(230, 40%, 80%);
    text-shadow:
      0 0 1em hsla(320, 100%, 50%, 0.2),
      0 0 0.125em hsla(320, 100%, 60%, 0.3),
      1em -0.125em 0.5em hsla(40, 100%, 60%, 0),
      -1em 0.125em 0.5em hsla(200, 100%, 60%, 0);
  }
`;

const titleModifiers = {
  animation: () => css`
    animation: ${lights} 5s 750ms linear infinite;
  `,

  normal: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,

  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.big};
  `,
};

export const Title = styled.h1<TitleProps>`
  ${({ theme, hasAnimation, textAlign, size }) => css`
    color: ${theme.colors.whiteText};
    font-family: ${theme.font.family.primary};
    font-weight: ${theme.font.medium};
    text-align: ${textAlign};

    ${hasAnimation && titleModifiers.animation()}
    ${titleModifiers[size!](theme)}

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xxbig};
    `}
  `}
`;
