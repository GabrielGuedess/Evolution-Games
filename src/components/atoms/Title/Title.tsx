import * as S from './Title.styles';

export type TitleProps = {
  children: React.ReactNode;
  textAlign?: 'center' | 'left';
};

export const Title = ({ children, textAlign = 'center' }: TitleProps) => (
  <S.Title textAlign={textAlign}>{children}</S.Title>
);
