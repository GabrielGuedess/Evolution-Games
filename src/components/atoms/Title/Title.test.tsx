import { renderWithTheme } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';
import 'jest-styled-components';

import theme from 'styles/theme';

import { Title } from './Title';

describe('<Title />', () => {
  it('should render the <Title /> component at its default size and center-aligned', () => {
    const { container } = renderWithTheme(<Title>Bestsellers</Title>);
    // Arrange
    const heading = screen.getByRole('heading', {
      name: 'Bestsellers',
    });

    // Assert
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveStyle({ 'font-size': '1.6rem' });
    expect(heading).toHaveStyle({ 'text-align': 'center' });
    expect(heading).toHaveStyleRule('font-size', '2.8rem', {
      media: '(min-width: 768px)',
    });

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the <Title /> component left-aligned', () => {
    renderWithTheme(<Title textAlign="left">Bestsellers</Title>);
    // Arrange
    const heading = screen.getByRole('heading', {
      name: 'Bestsellers',
    });

    // Assert
    expect(heading).toHaveStyle({ 'text-align': 'left' });
  });

  it('should render the <Title /> component text large', () => {
    renderWithTheme(<Title size="large">Bestsellers</Title>);
    // Arrange
    const heading = screen.getByRole('heading', {
      name: 'Bestsellers',
    });

    // Assert
    expect(heading).toHaveStyle({
      'font-size': theme.font.sizes.big,
    });
  });
});
