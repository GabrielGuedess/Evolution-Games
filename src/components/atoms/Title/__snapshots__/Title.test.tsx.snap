// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  '<Title /> should render the <Title /> component at its default size and center-aligned 1'
] = `
.c0 {
  color: #F4F4F5;
  font-size: 1.6rem;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-weight: 500;
  text-align: center;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2.8rem;
  }
}

<h1
  class="c0"
>
  Bestsellers
</h1>
`;
